backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/scheduling/utils/VdsCpuUnitPinningHelper.java

```diff
@@ -50,25 +45,16 @@ public boolean isDedicatedCpuPinningPossibleAtHost(Map<Guid, List<VdsCpuUnit>> v
 
         previewPinOfPendingExclusiveCpus(cpuTopology, vmToPendingPinnings, vm.getCpuPinningPolicy());
 
-        int vcpus = vm.getNumOfCpus();
-        int cpusLeft = vcpus;
-        int hostCoresPerSocket = host.getCpuCores() / host.getCpuSockets();
-        int hostThreadsPerCore = host.getCpuThreads() / host.getCpuCores();
+        int socketsLeft = vm.getNumOfSockets();
 
         for (int socket : getOnlineSockets(cpuTopology)) {
-            int coresInSockets = getCoresInSocket(cpuTopology, socket).size();
-            if (hostCoresPerSocket * hostThreadsPerCore == coresInSockets &&
-                    coresInSockets >= vcpus &&
-                    getFreeCpusInSocket(cpuTopology, socket).size() < vcpus) {
-                // we should fit into one socket
-                continue;
-            }
-            cpusLeft -= getFreeCpusInSocket(cpuTopology, socket).size();
-            if (cpusLeft <= 0) {
+            int coresInSocket = getAvailableCores(cpuTopology, socket, vm.getThreadsPerCpu());
+            socketsLeft -= coresInSocket / vm.getCpuPerSocket();
+            if (socketsLeft <= 0) {
                 return true;
             }
         }
-        return cpusLeft == 0;
+        return socketsLeft <= 0;
```

Comment:
can we return false here?

Reply:
no, why would we?

Reply:
maybe the diff-view here fools me again - but from what I see here, it's initialized to a positive number and we always check if it's <= 0 after reducing it so I don't see in which flow we can ever return true here

Reply:
yup, looking on it again, you are right. since the loop always checks it after reducing, when reaching here it means `socketsLeft > 0`.